compiler/src/dmd/dtemplate.d

```diff
@@ -3841,10 +3841,20 @@ MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplateParameters* param
                         tp = (*parameters)[i];
                     else
                     {
+                        Loc loc;
+                        // use the first location of template parameters as we
+                        // don't have any other location set anywhere.
+                        // not completely accurate, but better than nothing.
+                        if (inferStart < parameters.dim)
```

Comment:
can this ever be false?

Reply:
inferStart has a few assignments in a few places in the code where it's assigned to `parameters.dim`, would prefer to keep this check because I have not analyzed all the different sources where this function may be called from.

Reply:
It might be right to have it be an assertion.

Those are set to make [inferStart..end] == [] I think.

Issue is whether there's going to be an assertion failure because it. These functions are the back-rooms of dmd

Reply:
I think this whole file really needs a refactoring, there is a lot of code duplication happening here.

Reply:
That need for refactoring is why we don't have named args